invenio_rdm_records/resources/serializers/schemas.py

```diff
@@ -79,3 +79,100 @@ def get_contributors(self, obj):
         return [
             c["person_or_org"]["name"] for c in obj["metadata"].get("contributors", [])
         ] or missing
+
+
+class DumperMixin(Schema):
+    """A mixin that adds post-dump and pre-dump processing functionality to the `Schema` class.
+
+    This mixin provides a way to modify the serialized or deserialized output of a schema after or before it has been dumped.
+    """
+
+    def __init__(self, dumpers=None, **kwargs):
+        """Initialize a new instance of the `DumperMixin` class.
+
+        :param dumpers: A list of `DumpProcessorMixin` instances used to modify the serialized output, defaults to None.
+        :type dumpers: list, optional
+        :raises AssertionError: If any item in the `dumpers` list is not an instance of `DumpProcessorMixin`.
+        """
+        super().__init__(**kwargs)
+        self.dumpers = dumpers or []
+        assert all(isinstance(p, FieldDumper) for p in self.dumpers)
+
+    @post_dump(pass_original=True)
+    def post_dump_pipeline(self, data, original, many, **kwargs):
+        """Applies a sequence of post-dump steps to the serialized data.
+
+        This method defines a pipeline consisting of `DumpProcessorMixin` instances that implement a `post_dump` method.
+
+        :param data: The result of serialization.
+        :param original: The original object that was serialized.
+        :param many: Whether the serialization was done on a collection of objects.
+
+        :returns: The result of the pipeline processing on the serialized data.
+        """
+        for dumper in self.dumpers:
+            # Data is assumed to be modified and returned by the dumper
+            data = dumper.post_dump(data, original)
+        return data
+
+    @pre_dump
+    def pre_dump_pipeline(self, data, many, **kwargs):
+        """Applies a sequence of pre-dump steps to the input data.
+
+        This method defines a pipeline consisting of `DumpProcessorMixin` instances that implement a `pre_dump` method.
+
+        :param data: The result of serialization.
+        :param many: Whether the serialization was done on a collection of objects.
+
+        :returns: The result of the pipeline processing on the serialized data.
+        """
+        for dumper in self.dumpers:
+            # Data is assumed to be modified and returned by the dumper
+            data = dumper.pre_dump(data)
+        return data
+
+
+class LoaderMixin(Schema):
```

Comment:
Not used by now but since it's the same as the Dumper counter part I decided to add it straightaway